pkg/cluster/loadbalancerprofile.go

```diff
@@ -20,7 +20,7 @@ import (
 const outboundRuleV4 = "outbound-rule-v4"
 
```

Comment:
with this file being kinda massive, I wonder if we should have a `loadbalancerprofile` package instead and have different files for creation, deletion, and reconciliation code. This would also benefit the tests, as we could separate them out based on their purpose. wdyt?

Reply:
So I am piggy backing off the manager for the required cluster doc and azure clients for much of this code.  I am not sure how I would go about pulling any of the manager methods out of the cluster package.

There are plenty of functions that don't need the manager, I could pull those out into pkg/util.

Thoughts?

Reply:
Pulling out the non manager functions would be a good start in helping to narrow the scope of this file. I like that idea.

Reply:
I just wrapped up the refactor to move non-manager functions out of this file.  I also refactored testing a bit.  If you have a moment let me know what you think.  Thanks!

Reply:
I broke this PR up and the new home for these functions is in `pkg/util/loadbalancer`, see https://github.com/Azure/ARO-RP/pull/3379